### DISCUSSION TITLE
Issue resolving export modules

### DISCUSSION THREAD

**Discussion by dimavs** [preactjs/preact-cli#1790] [Q&A] [answered]
I am using the latest preact, but it looks like a webpack issue to me. Let's say my folder structure is:

- app
  - foo
    - index.js
  - app.js
- bar
  - baz
    - index.js

where 

app/app.js
```js
import { foo } from './foo'
import { baz } from '../bar/baz'
...
```

app/foo/index.js
```js
const foo = () => { ... }
export { foo }
```

bar/baz/index.js

```js
const baz = () => { ... }
export { baz }
```

It fails to bundle with a warning "export 'baz' was not found in ../bar/baz

But, if I make it a default export like

bar/baz/index.js

```js
const baz = () => { ... }
export default baz
```

and update app.js to

app/app.js
```js
import { foo } from './foo'
import baz from '../bar/baz'
...
```

everything works fine

Just wondering how is it possible that "local" directory reference resolves fine, but level up - can't be resolved in a list export.

**Comment by 6r17** [comment, marked as answer]
If I remember correctly there are differences in import behavior depending if you are importing a route or a component. But I could completely mislead you as I haven't played with Preact in a few months.

**Reply by dimavs** [reply]
Hm, weird, thanks. Yep, it happens to the route components, but they are still components. I'll dig a bit further.

**Reply by rschristian** [reply]
If you're in a `routes/` folder, you're falling prey to automatic code splitting. Routes are automatically split and need to provide default exports.
